src/collective/cover/browser/contentchooser.py

```diff
@@ -98,16 +101,17 @@ def search(self, query=None, page=0, b_size=10, uids=None):
         catalog_query = {'sort_on': 'effective', 'sort_order': 'descending'}
         catalog_query['portal_type'] = searchable_types
 
-        if query:
-            catalog_query = {'SearchableText': u'{0}*'.format(safe_unicode(query))}
-
-        # XXX: not implemented, this is needed?
-        # if uids:
-        #     catalog_query['UID'] = uids
-
+        query = safe_unicode(query)
         results = catalog(**catalog_query)
-        results = Batch(results, size=b_size, start=(page * b_size), orphan=0)
 
+        # let's filter using python
+        if query:
+            results = [
+                b for b in results
+                if (query.lower() in b['Title'].lower() or
+                    query.lower() in b['id'].lower())
+            ]
```

Comment:
If you test with collective.cover buildout there are some elements without title, and it becomes weird because you saw for example my-image there but when search for 'my-' didn't get the image